src/UI/Component/MainControls/Mainbar.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\MainControls;
+
+use ILIAS\UI\Component\Signal;
+use ILIAS\UI\Component\Button;
+use ILIAS\UI\Component\MainControls\Slate;
+use ILIAS\UI\Component\JavaScriptBindable;
+
+
+/**
+ * This describes the Mainbar
+ */
+interface Mainbar extends \ILIAS\UI\Component\Component, JavaScriptBindable
+{
+	/**
+	 * Append an entry.
+	 *
+	 * @param string $id
+	 * @param Button\Bulky|Slate $entry
+	 * @throws InvalidArgumentException 	if $id is already taken
+	 */
+	public function withEntry(string $id, $entry): Mainbar;
```

Comment:
Is it append or set? I guess append, so use **add**Entry.

Reply:
I think we had a similar discussion once and decided to use `withAdditional` in that case, because we expected people to think that just `add` would mutate the object.

Reply:
Then change to withAdditionalEntry.